elks/arch/i86/kernel/system.c

```diff
@@ -38,28 +39,33 @@ unsigned int INITPROC setup_arch(void)
      * Return start address for near heap allocator.
      */
 
+#ifdef SETUP_HEAPOFS
+    heapofs = (unsigned int)SETUP_HEAPOFS;
+#else
     /* Start heap allocations at even addresses */
-    endbss = (unsigned int)(_endbss + 1) & ~1;
+    heapofs = (unsigned int)(_endbss + 1) & ~1;
+#endif
```

Comment:
Adding a dynamic heap offset start is a nice enhancement, but things are getting ifdef messy. Lets leave this in as you have it for now, but I think it's time to split this whole routine outside of system.c into something more readable.

I've made a note of it, and will improve later.